js/profile.mjs

```diff
@@ -13,44 +12,50 @@ const profileImagePostComp = document.querySelector(
 const modalBackdrop = document.querySelector('.modal-backdrop');
 
 import {
-  getSessionStorage,
-  setSessionStorage,
-  checkIfLoggedIn,
+  getLocalStorage,
+  setLocalStorage,
   getPosts,
   getUsers,
   uploadImageToContainer,
   updateProfileInfo,
   followProfile,
   unfollowProfile,
-  setFetchLimitURL,
 } from './utils.mjs';
 import { displayAllPosts } from './layout.mjs';
 
-const globalSStorage = getSessionStorage();
-// export let profileDisplayed = globalSStorage && globalSStorage.name;
+const globalLocalStorage = getLocalStorage();
 
+/**
+ * displays users profile image or displays default image.
+ * @example
+ * ```js
+ * // call function
+ * getProfileImage()
+ * // function will check for users avatar and display as image, if user doesn't have avatar then a default image will be displayed instead.
+ * ```
+ */
 export async function getProfileImage() {
-  const user = await getUsers(globalSStorage.name, '');
-  if (user.avatar) {
-    profileImagePostComp.src = user.avatar;
+  if (globalLocalStorage.avatar) {
+    profileImagePostComp.src = globalLocalStorage.avatar;
   } else {
-    profileImagePostComp.src =
-      'https://upload.wikimedia.org/wikipedia/commons/8/89/Portrait_Placeholder.png';
+    profileImagePostComp.src = '../images/profile_placeholder.png';
+    // profileImagePostComp.src =
+    //   'https://upload.wikimedia.org/wikipedia/commons/8/89/Portrait_Placeholder.png';
   }
 }
 
 window.addEventListener('DOMContentLoaded', () => {
-  if (globalSStorage) {
-    if (globalSStorage.name === globalSStorage.profileDisplayed) {
-      displayProfileInfo(globalSStorage.name);
+  if (globalLocalStorage) {
+    if (globalLocalStorage.name === globalLocalStorage.profileDisplayed) {
+      displayProfileInfo(globalLocalStorage.name);
       getProfileImage();
     } else {
-      displayProfileInfo(globalSStorage.profileDisplayed);
+      displayProfileInfo(globalLocalStorage.profileDisplayed);
+      getProfileImage();
     }
   }
 });
 
-// uploadImageToContainer(container, input);
 if (newBannerInput && newAvatarInput && editProfileForm) {
   newBannerInput.addEventListener('keyup', () => {
     setTimeout(() => {
```

Comment:
`setTimeout` has a limited number of use cases and is usually a sign that something that should be handled with `async/await` is not.

If you want to push a function to the call stack instead of executing it synchronously, an anonymous async function is best:

```js
await async () => {
  // not thread blocking
}
```

However I would question the need for this in the first place. I suspect your intention here is to debounce or soften the blow on the call stack but not sure.